src/sundials/sundials_errors.c

```diff
@@ -77,10 +75,38 @@ void SUNAbortErrHandlerFn(int line, const char* func, const char* file,
                           const char* msg, SUNErrCode err_code,
                           void* err_user_data, SUNContext sunctx)
 {
-  char* file_and_line = combineFileAndLine(line, file);
+  char* file_and_line = sunCombineFileAndLine(line, file);
   SUNLogger_QueueMsg(sunctx->logger, SUN_LOGLEVEL_ERROR, file_and_line, func,
                      "SUNAbortErrHandler: Calling abort now, use a different "
                      "error handler to avoid program termination.\n");
   free(file_and_line);
   abort();
 }
+
+void SUNGlobalFallbackErrHandler(int line, const char* func, const char* file,
+                                 const char* msgfmt, SUNErrCode err_code, ...)
+{
+  va_list ap;
+  char* log_msg       = NULL;
+  char* file_and_line = NULL;
+
+  va_start(ap, err_code);
+
+  file_and_line = sunCombineFileAndLine(__LINE__, __FILE__);
+  sunCreateLogMessage(SUN_LOGLEVEL_ERROR, 0, file_and_line,
+                      __func__, "The SUNDIALS SUNContext was corrupt or NULL when an error occurred. As such, error messages have been printed to stderr.",
```

Comment:
Is it really *always* the case that if `SUNGlobalFallbackErrHandler` is called, then the SUNDIALS SUNContext was corrupt or NULL when the error occurred, or is it possible to use this fallback function just because no other error handlers were specified?  If this could be used in any other situation, then I recommend that this line be adjusted.

Reply:
It will only ever be called if the `SUNContext` is `NULL` or the object holding the `SUNContext` is `NULL` and any use of it otherwise should be avoided. If there is no error handler attached, errors will be ignored (by design).